libs/vgc/ui/action.h

```diff
@@ -66,8 +88,55 @@ class VGC_UI_API Action : public core::Object {
         return shortcut_;
     }
 
+    /// Sets the shortcut associated with this action.
+    ///
+    void setShortcut(const Shortcut& shortcut) {
+        shortcut_ = shortcut;
+        changed().emit(this);
+    }
+
+    VGC_SIGNAL(disabled, (Action*, action))
+
+    /// Returns whether this action is disabled or not.
+    ///
+    bool isDisabled() const {
+        return isDisabled_;
+    }
+
+    /// Sets the disabled state of this action.
+    ///
+    void setDisabled(bool disabled) {
+        isDisabled_ = disabled;
+        this->disabled().emit(this);
```

Comment:
I don't think we should emit disabled if `disabled == false`. I think it makes more sense to do what Qt does here: use `setEnabled` instead, and have a signal `enabledChanged`: https://doc.qt.io/qt-6/qaction.html#enabled-prop.